src/main/java/net/dries007/tfc/objects/recipes/UnmoldRecipe.java

```diff
@@ -5,182 +5,172 @@
 
 package net.dries007.tfc.objects.recipes;
 
+import java.util.List;
 import javax.annotation.Nonnull;
 import javax.annotation.ParametersAreNonnullByDefault;
 
+import com.google.common.collect.Lists;
+import com.google.gson.JsonObject;
+import net.minecraft.client.util.RecipeItemHelper;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.inventory.InventoryCrafting;
 import net.minecraft.item.ItemStack;
 import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.item.crafting.Ingredient;
+import net.minecraft.util.JsonUtils;
 import net.minecraft.util.NonNullList;
+import net.minecraft.util.ResourceLocation;
 import net.minecraft.world.World;
 import net.minecraftforge.common.ForgeHooks;
-import net.minecraftforge.fluids.capability.IFluidHandler;
-import net.minecraftforge.registries.IForgeRegistryEntry;
+import net.minecraftforge.common.crafting.IRecipeFactory;
+import net.minecraftforge.common.crafting.JsonContext;
+import net.minecraftforge.oredict.ShapelessOreRecipe;
 
 import net.dries007.tfc.Constants;
 import net.dries007.tfc.api.capability.IMoldHandler;
 import net.dries007.tfc.api.capability.heat.IItemHeat;
 import net.dries007.tfc.api.types.Metal;
-import net.dries007.tfc.objects.items.ceramics.ItemMold;
 import net.dries007.tfc.objects.items.metal.ItemMetal;
 
 import static net.dries007.tfc.api.capability.heat.CapabilityItemHeat.ITEM_HEAT_CAPABILITY;
-import static net.dries007.tfc.api.util.TFCConstants.MOD_ID;
 import static net.minecraftforge.fluids.capability.CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY;
 
 @ParametersAreNonnullByDefault
-public class UnmoldRecipe extends IForgeRegistryEntry.Impl<IRecipe> implements IRecipe
+public class UnmoldRecipe extends ShapelessOreRecipe
 {
-    private final ItemMold mold;
+    int moldSlotID;
```

Comment:
Creating global variables to hold state between function calls is awful practice here. You have no guarantee that the `IRecipe` methods are called in a specific order, and this is prone to all sorts of horrible errors. If you need to check state in multiple methods, then check the state again - do not rely on whatever was passed in in a different method call.

Reply:
Okay, I see